src/timely-util/src/containers/stack.rs

```diff
@@ -169,6 +167,30 @@ impl<T: Clone + Columnation + 'static> Container for StackWrapper<T> {
     }
 }
 
+impl<T: Clone + Columnation + 'static> SizableContainer for StackWrapper<T> {
+    fn capacity(&self) -> usize {
+        match self {
+            StackWrapper::Legacy(l) => l.capacity(),
+            StackWrapper::Chunked(c) => c.capacity(),
+        }
+    }
+
+    fn preferred_capacity() -> usize {
+        if ENABLE_CHUNKED_STACK.load(std::sync::atomic::Ordering::Relaxed) {
```

Comment:
You or I should check whether the two branches return different values. I think they should not because changing the preferred capacity at run time is a case I didn't think about and can cause undesired effects.

Reply:
They both end up in `timely_container::buffer::preferred_capacity`

Reply:
Sorry, I've intended to make this a stronger statement: It should just call `timely::container::buffer::preferred_capacity` to protect against someone changing the preferred capacity and suddenly things starting to crash/oom. `preferred_capacity` should probably be a `const` fn.

Reply:
No worries, I will put up a new PR to fix this